application/src/main/java/org/opentripplanner/street/search/request/TimeSlopeSafetyTriangle.java

```diff
@@ -0,0 +1,189 @@
+package org.opentripplanner.street.search.request;
+
+import static org.opentripplanner.utils.lang.DoubleUtils.doubleEquals;
+import static org.opentripplanner.utils.lang.DoubleUtils.roundTo2Decimals;
+
+import java.util.Objects;
+import org.opentripplanner.utils.tostring.ValueObjectToStringBuilder;
+
+/**
+ * Sets the (bicycle or scooter) triangle routing parameters -- the relative importance of safety,
+ * flatness, and speed. These three fields should have values between 0 and 1, and should add up to
+ * 1.
+ * <p>
+ * THIS CLASS IS IMMUTABLE AND THREAD-SAFE.
+ */
+public final class TimeSlopeSafetyTriangle {
+
+  private static final double ZERO = 0.0;
+
+  public static final TimeSlopeSafetyTriangle DEFAULT = new TimeSlopeSafetyTriangle(1, 1, 1);
+  private final double time;
+  private final double slope;
+  private final double safety;
+
+  /**
+   * Sets the bicycle or scooter triangle routing parameters -- the relative importance of safety,
+   * flatness, and speed. These three fields of the RouteRequest should have values between 0 and 1,
+   * and should add up to 1. This setter function accepts any three numbers and will normalize them
+   * to add up to 1.
+   */
+  private TimeSlopeSafetyTriangle(double time, double slope, double safety) {
+    safety = positiveValueOrZero(safety);
+    slope = positiveValueOrZero(slope);
+    time = positiveValueOrZero(time);
+
+    if (zeroVector(time, slope, safety)) {
+      time = 1.0;
+      slope = 1.0;
+      safety = 1.0;
+    }
+
+    // Normalize
+    double total = safety + slope + time;
+    time /= total;
+    slope /= total;
+
+    // We round to closest 2 decimals
+    this.time = roundTo2Decimals(time);
+    this.slope = roundTo2Decimals(slope);
+    this.safety = roundTo2Decimals(1.0 - (this.time + this.slope));
+  }
+
+  /**
+   * Creates a copy of this object (with the same values) as a builder.
+   */
+  public Builder copyOf() {
```

Comment:
I added some Javadoc, but I don't believe that it's the same as `of()` since it's non-static.